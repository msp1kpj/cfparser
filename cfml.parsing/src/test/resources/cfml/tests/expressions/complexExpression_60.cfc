component{

   function name(){
    var params = {
      thingId = { "value" = thing.getID(), "cfsqltype" = "CF_SQL_INTEGER" },
      name = { "value" = thing.getName(), "cfsqltype" = "CF_SQL_VARCHAR", null = !len( thing.getName() ) }
    };
   }
}