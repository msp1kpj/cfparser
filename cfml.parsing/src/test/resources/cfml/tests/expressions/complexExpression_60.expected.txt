/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
FUNCTION                 <function>
IDENTIFIER               <name>
'('                      <(>
')'                      <)>
'{'                      <{>
VAR                      <var>
IDENTIFIER               <params>
'='                      <=>
'{'                      <{>
IDENTIFIER               <thingId>
'='                      <=>
'{'                      <{>
OPEN_STRING              <">
STRING_LITERAL           <value>
CLOSE_STRING             <">
'='                      <=>
IDENTIFIER               <thing>
'.'                      <.>
IDENTIFIER               <getID>
'('                      <(>
')'                      <)>
','                      <,>
OPEN_STRING              <">
STRING_LITERAL           <cfsqltype>
CLOSE_STRING             <">
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <CF_SQL_INTEGER>
CLOSE_STRING             <">
'}'                      <}>
','                      <,>
IDENTIFIER               <name>
'='                      <=>
'{'                      <{>
OPEN_STRING              <">
STRING_LITERAL           <value>
CLOSE_STRING             <">
'='                      <=>
IDENTIFIER               <thing>
'.'                      <.>
IDENTIFIER               <getName>
'('                      <(>
')'                      <)>
','                      <,>
OPEN_STRING              <">
STRING_LITERAL           <cfsqltype>
CLOSE_STRING             <">
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <CF_SQL_VARCHAR>
CLOSE_STRING             <">
','                      <,>
IDENTIFIER               <null>
'='                      <=>
'!'                      <!>
IDENTIFIER               <len>
'('                      <(>
IDENTIFIER               <thing>
'.'                      <.>
IDENTIFIER               <getName>
'('                      <(>
')'                      <)>
')'                      <)>
'}'                      <}>
'}'                      <}>
';'                      <;>
'}'                      <}>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          function
          (identifier name)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (localAssignmentExpression
                var
                (multipartIdentifier (identifier params))
                =
                (startExpression
                  (compareExpression
                    (baseExpression
                      (unaryExpression
                        (primaryExpression
                          (implicitStruct
                            {
                            (implicitStructElements
                              (implicitStructExpression
                                (implicitStructKeyExpression (multipartIdentifier (identifier thingId)))
                                =
                                (baseExpression
                                  (unaryExpression
                                    (primaryExpression
                                      (implicitStruct
                                        {
                                        (implicitStructElements
                                          (implicitStructExpression
                                            (implicitStructKeyExpression
                                              (literalExpression (stringLiteral " (stringLiteralPart value) "))
                                            )
                                            =
                                            (baseExpression
                                              (unaryExpression
                                                (memberExpression
                                                  (identifier thing)
                                                  .
                                                  (functionCall (identifierOrReservedWord (identifier getID)) ( argumentList ))
                                                )
                                              )
                                            )
                                          )
                                          ,
                                          (implicitStructExpression
                                            (implicitStructKeyExpression
                                              (literalExpression (stringLiteral " (stringLiteralPart cfsqltype) "))
                                            )
                                            =
                                            (baseExpression
                                              (unaryExpression
                                                (primaryExpression
                                                  (literalExpression (stringLiteral " (stringLiteralPart CF_SQL_INTEGER) "))
                                                )
                                              )
                                            )
                                          )
                                        )
                                        }
                                      )
                                    )
                                  )
                                )
                              )
                              ,
                              (implicitStructExpression
                                (implicitStructKeyExpression (multipartIdentifier (identifier name)))
                                =
                                (baseExpression
                                  (unaryExpression
                                    (primaryExpression
                                      (implicitStruct
                                        {
                                        (implicitStructElements
                                          (implicitStructExpression
                                            (implicitStructKeyExpression
                                              (literalExpression (stringLiteral " (stringLiteralPart value) "))
                                            )
                                            =
                                            (baseExpression
                                              (unaryExpression
                                                (memberExpression
                                                  (identifier thing)
                                                  .
                                                  (functionCall (identifierOrReservedWord (identifier getName)) ( argumentList ))
                                                )
                                              )
                                            )
                                          )
                                          ,
                                          (implicitStructExpression
                                            (implicitStructKeyExpression
                                              (literalExpression (stringLiteral " (stringLiteralPart cfsqltype) "))
                                            )
                                            =
                                            (baseExpression
                                              (unaryExpression
                                                (primaryExpression
                                                  (literalExpression (stringLiteral " (stringLiteralPart CF_SQL_VARCHAR) "))
                                                )
                                              )
                                            )
                                          )
                                          ,
                                          (implicitStructExpression
                                            (implicitStructKeyExpression (multipartIdentifier (identifier null)))
                                            =
                                            (baseExpression
                                              (notExpression
                                                !
                                                (startExpression
                                                  (compareExpression
                                                    (baseExpression
                                                      (unaryExpression
                                                        (memberExpression
                                                          (functionCall
                                                            (identifierOrReservedWord (identifier len))
                                                            (
                                                            (argumentList
                                                              (argument
                                                                (startExpression
                                                                  (compareExpression
                                                                    (baseExpression
                                                                      (unaryExpression
                                                                        (memberExpression
                                                                          (identifier thing)
                                                                          .
                                                                          (functionCall (identifierOrReservedWord (identifier getName)) ( argumentList ))
                                                                        )
                                                                      )
                                                                    )
                                                                  )
                                                                )
                                                              )
                                                            )
                                                            )
                                                          )
                                                        )
                                                      )
                                                    )
                                                  )
                                                )
                                              )
                                            )
                                          )
                                        )
                                        }
                                      )
                                    )
                                  )
                                )
                              )
                            )
                            }
                          )
                        )
                      )
                    )
                  )
                )
              )
              ;
            )
            }
          )
        )
      )
      }
    )
  )
)
/*======*/